plugins/c8y_configuration_plugin/src/main.rs

```diff
@@ -135,60 +142,87 @@ async fn run(
     let () = mqtt_client.published.send(msg).await?;
 
     // Mqtt message loop
+    process_mqtt_messages(
+        &mut plugin_config,
+        &mut mqtt_client,
+        config_file_path,
+        http_client,
+        tmp_dir,
+    )
+    .await?;
+
+    mqtt_client.close().await;
+
+    Ok(())
+}
+
+async fn process_mqtt_messages(
+    plugin_config: &mut PluginConfig,
+    mqtt_client: &mut Connection,
+    config_file_path: &Path,
+    http_client: &mut impl C8YHttpProxy,
+    tmp_dir: PathBuf,
+) -> Result<(), anyhow::Error> {
     while let Some(message) = mqtt_client.received.next().await {
         debug!("Received {:?}", message);
         if let Ok(payload) = message.payload_str() {
-            let result = if let "tedge/configuration_change/c8y-configuration-plugin" =
-                message.topic.name.as_str()
-            {
-                // Reload the plugin config file
-                plugin_config = PluginConfig::new(config_file_path);
-                // Resend the supported config types
-                let msg = plugin_config.to_supported_config_types_message()?;
-                mqtt_client.published.send(msg).await?;
-                Ok(())
-            } else {
-                match payload.split(',').next().unwrap_or_default() {
-                    "524" => {
-                        let maybe_config_download_request =
-                            SmartRestConfigDownloadRequest::from_smartrest(payload);
-                        if let Ok(config_download_request) = maybe_config_download_request {
-                            handle_config_download_request(
-                                &plugin_config,
-                                config_download_request,
-                                tmp_dir.clone(),
-                                &mut mqtt_client,
-                                http_client,
-                            )
-                            .await
-                        } else {
-                            error!("Incorrect Download SmartREST payload: {}", payload);
-                            Ok(())
+            let result = match message.topic.name.as_str() {
+                "tedge/configuration_change/c8y-configuration-plugin" => {
+                    // Reload the plugin config file
+                    let plugin_config = PluginConfig::new(config_file_path);
+                    // Resend the supported config types
+                    let msg = plugin_config.to_supported_config_types_message()?;
+                    mqtt_client.published.send(msg).await?;
+                    Ok(())
+                }
+                "tedge/health-check" | "tedge/health-check/c8y-configuration-plugin" => {
```

Comment:
This needs to be replaced by a `if health_check_topics.accept(&message)`

Reply:
Fixed it if health_check_topics("c8y-configuration-plugin").accept(&message)